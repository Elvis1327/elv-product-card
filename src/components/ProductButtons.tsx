import React from 'react';
import { CSSProperties, useCallback, useContext } from "react";

import { ProductContext } from "./ProductCard";
import styles from '../styles/styles.module.css';

interface Props {
    className?: string;
    optionColor?: string;
    style?: CSSProperties;
}

export const ProductButtons = ({className, optionColor, style}: Props) => {
    const { counter, increaseBy, maxCount } = useContext(ProductContext); 
    
    const isMaxReached = useCallback(() => !!maxCount && counter === maxCount,
        [counter, maxCount],
    )


    return (
        <div className={styles.buttonsContainer} style={style}>
            <button
                className={`${styles.buttonMinus} ${optionColor}`}
                onClick={() => increaseBy(-1)}
            >
                    -
            </button>
            <div
                className={styles.countLabel}
            >
                {counter}
            </div>
            <button
                className={`${styles.buttonAdd} ${ isMaxReached() && styles.disabled} ${className}`}
                onClick={() => increaseBy(1)}
            >
                +
            </button>
        </div>
    )
};