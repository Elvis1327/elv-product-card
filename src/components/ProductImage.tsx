import React from 'react';
import { useContext, CSSProperties } from "react";

import noImage from '../assets/no-image.jpg';
import { ProductContext } from "./ProductCard";
import styles from '../styles/styles.module.css';

interface Props {
    img?:string;
    className?: string;
    style?: CSSProperties;
}

export const ProductImage = ({img, className, style}: Props) => {
    const { product } = useContext(ProductContext);

    let imgToShow: string = '';
    if(img) {
        imgToShow = img;
    }else if(product.img){
        imgToShow = product.img
    }else{
        imgToShow = noImage
    }

    return (
        <img 
                className={`${styles.productImg} ${className}`}
                src={imgToShow} 
                alt="img-pic"
                style={style}
            />
    )
};