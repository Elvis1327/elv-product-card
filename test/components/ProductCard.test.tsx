import React from 'react';
import renderer from 'react-test-renderer';

import { ProductCard } from '../../src/components/index';
import { product1 } from '../data/products';

describe('ProductCard', () => {

    test('Debe de mostrar el comopnente Correctamente', () => {
        const wrapper = renderer.create(
            <ProductCard
                product={product1}
            >
                {
                    () => (
                        <h1>Proudct Card</h1>
                    )
                }
            </ProductCard>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('debe de incrementer el contador', () => {
        const wrapper = renderer.create(
            <ProductCard
                product={product1}
            >
                {
                    ({count, increaseBy}) => (
                        <>
                            <h1>Porudct Card</h1>
                            <span>
                                {count}
                            </span>
                            <button
                                onClick={ () => increaseBy(1) }
                            >

                            </button>
                        </>
                    )
                }
            </ProductCard>
        );
        let tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot();
    })

})
