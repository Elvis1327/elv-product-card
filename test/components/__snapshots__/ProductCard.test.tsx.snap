// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Debe de mostrar el comopnente Correctamente 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Proudct Card
  </h1>
</div>
`;

exports[`ProductCard debe de incrementer el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Porudct Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;
